README

```diff
@@ -143,7 +145,7 @@ General notes
 Platform Notes
 --------------
 
-- N/A
+- Mac OS X versions prior to 10.5.0 are not supported on 64-bit systems.
```

Comment:
Per this chunk of configure.ac, I think you don't need this README note:

https://github.com/open-mpi/ompi/blob/8b228c1314dc81c799e14b9988b7b0c41d125f8a/configure.ac#L328-L347

Reply:
Oops, missed that.  I removed the README change and updated the commit message to match this new reality.